content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md

```diff
@@ -123,19 +123,44 @@ by using a list of well known domain socket paths. To use different container ru
 if there are more than one installed on the provisioned node, specify the `--cri-socket`
 argument to `kubeadm init`. See [Installing runtime](/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#installing-runtime).
 1. (Optional) Unless otherwise specified, kubeadm uses the network interface associated
-with the default gateway to advertise the master's IP. To use a different
+with the default gateway to advertise the control-plane's IP. To use a different
 network interface, specify the `--apiserver-advertise-address=<ip-address>` argument
 to `kubeadm init`. To deploy an IPv6 Kubernetes cluster using IPv6 addressing, you
 must specify an IPv6 address, for example `--apiserver-advertise-address=fd00::101`
 1. (Optional) Run `kubeadm config images pull` prior to `kubeadm init` to verify
-connectivity to gcr.io registries.   
+connectivity to gcr.io registries.
 
-Now run:
+To initialize the control-plane node run:
 
 ```bash
 kubeadm init <args>
 ```
 
+### Considerations about ControlPlaneEndpoint
+
+Instead of using `--apiserver-advertise-address`, `kubeadm init` also allows you to use the `--control-plane-endpoint` flag.
+While `--apiserver-advertise-address` can be used to set the advertise address for this particular control-plane node's API
+server, `--control-plane-endpoint` is preferred and can be used to set the shared endpoint for all control-plane nodes.
+Such an endpoint can be either a DNS name or an IP address of a load-balancer.
+
+If you have plans to upgrade this single control-plane kubeadm cluster to high-availability you should definitely use
+`--control-plane-endpoint`. This will allow you to pass any DNS name and later change the address that the DNS name resolves too.
+Without installing a DNS server and utilizing `A` and `CNAME` records, you can achieve this abstraction by adding an entry
+in `/etc/hosts` for all your nodes. For example:
```

Comment:
I'm not sure it's a good idea to add instructions for doing this via `/etc/hosts` and should instead push users towards DNS or LB-based solutions.

Reply:
i'd agree, any preference for a Linux DNS server that we should provide a minimal example for?

Reply:
If all we care about is local resolution as a stopgap, then dnsmasq wouldn't be a bad option (and integrates nicely with NetworkManager)

Reply:
ok, i will provide an example for dnsmasq instead.

Reply:
after some research and experiment, it seems the situation with dnsmasq is a bit messy on Ubuntu with `systemd-resolved` (and with NetworkManager according to this ticket https://bugs.launchpad.net/ubuntu/+source/network-manager/+bug/959037)

installing DNS servers like Bind9 can also conflict with existing software by the distros, so it feels like recommending DNS severs will just make the users enter a rabbit hole from our docs.

`/etc/hosts` is something that *should* work everywhere, but i think i'm just going to remove the `/etc/hosts` mention and just say something like "use your preferred hostname-to-IP mapping solution".